tests/testthat/test-sample_lognormal.R

```diff
@@ -0,0 +1,73 @@
+# Instructions ----
+#' This file follows the format generated by ecosystemdata:::use_testthat_template().
+#' Necessary tests include input and output (IO) correctness [IO
+#' correctness], edge-case handling [Edge handling], and built-in errors and
+#' warnings [Error handling]. See `?ecosystemdata:::use_testthat_template` for more
+#' information. Every test should have a @description tag that takes up just
+#' one line, which will be used in the bookdown report of {testthat} results.
+
+# Setup ----
+# Load or prepare any necessary data for testing
+
+# sample_lognormal ----
+## IO correctness ----
+test_that("sample_lognormal() works with correct inputs", {
+  #' @description Test that sample_lognormal(1, 1) returns known result when
+  #' the seed is set to a defined value of 123.
+  set.seed(123)
+  expect_equal(
+    object = sample_lognormal(1, 1),
+    expected = 0.346291059
+  )
+
+  #' @description Test that sample_lognormal(x) returns two different values
+  #' when the seed is not set.
+  expect_false(
+    sample_lognormal(1) == sample_lognormal(1)
+  )
+})
+
+## Edge handling ----
+test_that("sample_lognormal() returns correct outputs for edge cases", {
+  #' @description Test that sample_lognormal(x) returns a vector when a single
+  #' value is passed to `x` but a vector is passed to `sd`.
+  expect_equal(
+    object = length(sample_lognormal(1:5, 1)),
+    expected = 5
+  )
+
+  #' @description Test that sample_lognormal(x) returns a vector when a single
+  #' value is passed to `sd` but a vector is passed to `x`.
+  expect_equal(
```

Comment:
Question: If I understand correctly, the user provides a single observation value and a vector of sds. The `stats::rnorm()`  samples one value using only the first `sd`. However, `bias_correction` returns 5 values because the `sd` vector has 5 elements. As a result, the final expected values also have a length of 5. Should we replicate `x` into a vector of length 5 before using `stats::rnorm()`, or should we instead use only the first `sd` when calculating `bias_correction`? Or am I misunderstanding something here?